### ISSUE TITLE
DEADLINE_EXCEEDED when uploading results

### ISSUE THREAD

**Issue by FFrostYY** [waymo-research/waymo-open-dataset#183]
Hello,
I tried to evaluate an existing object detection model on the validation set for my thesis work. I uploaded the file to the evaluation server, and after ~50 minutes got the following error:
` DEADLINE_EXCEEDED: [type.googleapis.com/util.ErrorSpacePayload='RPC::DEADLINE_EXCEEDED'] `

What could be the cause of this? The submission link is: https://waymo.com/open/challenges/entry/?timestamp=1590842929486681&challenge=DETECTION_2D&email=gusfatfi@student.gu.se&submissionType=VALIDATION

**Comment by peisun1115** [comment]
Did you limit the number of objects per frame? 400 per frame is good enough. If you need more, you can try, it should work but not something like several thousands per frame.

**Comment by FFrostYY** [comment]
Thanks for the quick response. I will try limiting the number of objects and close this issue once I've confirmed it works.

**Comment by FFrostYY** [comment]
@peisun1115 This did not resolve the issue. I added and tested a limit to my code that skips the frame once 400 objects are exceeded, but still getting the same error.

**Comment by shinya7y** [comment]
Hi @peisun1115 ,
I also got the error in a submission for Test Set, though the same file can be uploaded for Validation Set.
Is the number of DEADLINE_EXCEEDED submissions counted in the "3 times within 30 days" rule?
Participants cannot assess whether the server is overloaded, and of cource, cannot do test set tuning with DEADLINE_EXCEEDED.

**Comment by peisun1115** [comment]
usually this is caused by something else. I will take a look soon.

Any errors in the test set won't be counted in "3 times within 30 days" rule?

**Comment by peisun1115** [comment]
@FFrostYY 

For your case, it is because all of your predictions (at least shard 0 which i have examined) are from one scene 967082162553397800_5102_900_5122_900. We shard by scene. So it took too long to process this shard. That is why it is DEADLINE_EXCEEDED.

**Comment by FFrostYY** [comment]
@peisun1115 I see, I am closing this issue since the reason has been found. Is it possible to evaluate only a part of the validation set in any way? I was using one scene to test my data format, and evaluating on the entire set took far too long.

**Comment by peisun1115** [comment]
The root cause is that you have too many predictions per frame within this scene. I parsed your file shard 1. here are the top number of predictions per frame (1 column is count, 2 column is the frame_timestamp_micros)

   2357 1507598025051394
   2151 1507598016950946
   2138 1507598017050903
   1941 1507598024951431
   1793 1507598025151360
   1692 1507598027650975
   1692 1507598024851435
   1691 1507598016850980
   1662 1507598017150869
   1617 1507598025351287

**Comment by peisun1115** [comment]
You can eval part of the dataset by running our compute_detection_metrics tool locally or use the tensorflow op api we provided.

**Comment by shinya7y** [comment]
@peisun1115 Thank you for your quick reply! I have cared about only average number of predictions.

**Comment by DezeZhao** [comment]
@peisun1115 hello, I meet this problem, DEADLINE_EXCEEDED: [type.googleapis.com/util.ErrorSpacePayload='RPC::DEADLINE_EXCEEDED'] could you pls help me?

**Comment by DezeZhao** [comment]
on validation server

**Comment by DezeZhao** [comment]
oh , I find the cause! Excuse me!
